### ISSUE TITLE
Post Response Script environment variable

### ISSUE THREAD

**Issue by federicorosso1993** [usebruno/bruno#556]
I'm trying to migrate from postman to bruno and I need some for the post response script

I'm using the bruno desktop UI with this code

`bru.setVar("accessToken", res.body.payload["access_token"]);`

I don't really know if res.body.payload["access_token"] is the correct way to access my response
but before that I notice that even doing something as easy as 

`bru.setVar("accessToken", "abc");`

do not work at all... (and I don't know why)
In my environment I already got my accessToken variable empty, and it stay empty (even in the file if I try to open it so it is not an interface visualization error).
I also tried with var tab (that I don't think I will need since I'm trying to set the variable on the enviroment) but it still will not work (the variable on the var tab stay empty).

I already notice that I cannot import postman environment (at least bruno give me an error when I try, if you know a way please tell me that also) so if this is not working I will have to continue with ARC for now (even if not supported anymore... and I already like bruno since it can import from postman collection without any problem unlike ARC).

**Comment by n00o** [comment]
When you do `res.body` it gives you the response of the body in this case it is in JSON form so you can reference the descendent values in the same way. It would look like this.
```JavaScript
bru.setVar("accessToken",res.body.access_token)
```
If you wanted to use the variable tab you could just use the `res.body.access_token`. The example for this is found here: https://docs.usebruno.com/scripting/vars.html.

**Comment by federicorosso1993** [comment]
> > > When you do `res.body` it gives you the response of the body in this case it is in JSON form so you can reference the descendent values in the same way. It would look like this.
> > > ```
> > > bru.setVar("accessToken",res.body.access_token)
> > > ```
> > > 
> > > 
> > > If you wanted to use the Vars tab you could just use the `res.body.access_token`. The example for this is found here: https://docs.usebruno.com/scripting/vars.html.
> > 
> > 
> > I would like to save that value in an environment variable. How to do that?
> 
> Then use the Post Response Script and the function would be this instead.
> 
> ```js
> bru.setEnvVar("accessToken",res.body.access_token)
> ```
> 
> I don't think there is a way to use the Vars tab to assign Environment variables though. More information can be found here: https://docs.usebruno.com/scripting/javascript-reference.html#setenvvar. It does make sense to store authorization tokens inside of an environment so that switching environments would make you need to use a new authorization token.

Ok so right now I cannot set an environment variable like I did in postman or even ARC. 
Ok I think right now Bruno is not ready to replace postman, at least for me. 
The good news is that I found Hopp Scotch (thank to Bruno page where it show the current most used project... it is the only other one open source and free) and it is almost everything I need 
There I can set an env variable like this

`pw.env.set("accessToken", pw.response.body.payload.access_token);`

I don't like that it is not a standalone app (I tried the Tauri app but it will only be like a webview), and that I cannot just add my folder-files/collection to a local git repository (like I can with Bruno and like I could with Thunder Client), but it is better than ARC that it is not supported anymore... 

I will still keep a look for Bruno, maybe when it will be ready it will be the best, and it is still the way to go for my personal projects where I don't need to do lot's of testing.

**Comment by helloanoop** [comment]
Hey @federicorosso1993 I need to write some documentation on vars and hopefully make a video that will make things arounds a lot more clearer.

I look forward to hearing back again from you :)

Will keep this ticket open to make a good video on explaining vars in Bruno

**Comment by federicorosso1993** [comment]
> Hey @federicorosso1993 I need to write some documentation on vars and hopefully make a video that will make things arounds a lot more clearer.
> 
> I look forward to hearing back again from you :)
> 
> Will keep this ticket open to make a good video on explaining vars in Bruno

Thank you. I would also love if you could include in your tutorial if there is a way to call an endpoint before each call...
For example... I got a GET that will give me an access-token, it will be amazing to be able to call it before each call and use the (automatically because if the GET got a new access-token it will be automatically set to a var) updated access-token to do every other call. It is connected to the previous probem (since I was trying to set an environment variable to be able to access it everywhere in my collection) and it is one of the thing that Postman and Thunder Client can do, but Hopp Scotch for example cannot do (the post request script there can only set environment variable, but the pre request script cannot call my GET... not on every single call, not at collection level).

If Bruno is or will be able to do that I can switch to it. 
Today I was able to actually create a regular post request variable and get my access token like this 
`res.body.payload["access_token"]`
in Bruno... I can see it in the Variables view (didn't notice it yesterday), still not sure if I can use it in another call, I will try later.
If this is true maybe I don't need it at the environment level... but there can still be other case where I could need it.

**Comment by WilliamHuang1995** [comment]
Sorry to piggy back off this thread, but I am trying to figure out if bruno has something similar to offer wrt this image
![image](https://github.com/usebruno/bruno/assets/17118667/196a9b08-ace8-44e0-b120-9d66844f360a)


I currently have a request specifically to get the access token, and it will also resend request to retrieve when needed (since tokens time out)

Essentially
Request 1 - Fetch Token
Request 2 - Call Normal API

Put simply I want to invoke request 1 everytime I want to call Request 2
Is it currently possible?

**Comment by tbalaskovitz** [comment]
I have a similar issue. i just have a response that is a bearer_token and I'm trying for the life of me to set it to my environment variable, but it doesn't want to set.

**Comment by DaniaHamad** [comment]
Helloo,
Mayhaps this helps?

bru.setEnvVar("accessToken", res.body.data.accessToken);

**Comment by danielo515** [comment]
> Helloo, Mayhaps this helps?
> 
> bru.setEnvVar("accessToken", res.body.data.accessToken);

Sadly that does not help. Env vars are not set like that sadly

**Comment by sreelakshmi-bruno** [comment]
Closing this ticket since this issue has been fixed. Please reopen if you find otherwise, thanks!

**Comment by douglasdelarosa** [comment]
On visual studiio, this still happening
